### PR TITLE
Added meta map to module and tests

### LINKED ISSUE #5152: Add meta to fastme
### Is your feature request related to a problem? Please describe

Batch processing with this modules is difficult, as output files cannot be traced back to their origin.

### Describe the solution you'd like

Add meta map.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by itrujnara**
## PR checklist

Closes #5152 

- [x] Added meta map to input and output to facilitate batch processing.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`

### CODE REVIEW

**Review by JoseEspinosa — APPROVED** [review]
Just a small format suggestion, otherwise awesome! 🚀 
Feel free to merge once addressed

File: modules/nf-core/fastme/main.nf [hunk 1]
```diff
@@ -8,15 +8,14 @@ process FASTME {
...
-    path initial_tree
+    tuple val(meta), path(infile), path(initial_tree)
 
     output:
-    path "*.nwk"        , emit: nwk
-    path "*_stat.txt"   , emit: stats
+    tuple val(meta), path("*.nwk")        , emit: nwk
+    tuple val(meta), path("*_stat.txt")   , emit: stats
+    tuple val(meta), path("*.matrix.phy") , emit: matrix    , optional: true
+    tuple val(meta), path("*.bootstrap")  , emit: bootstrap , optional: true
```
Comment (JoseEspinosa) [inline modules/nf-core/fastme/main.nf:17]:
```suggestion
    tuple val(meta), path("*.nwk")       , emit: nwk
    tuple val(meta), path("*_stat.txt")  , emit: stats
    tuple val(meta), path("*.matrix.phy"), emit: matrix    , optional: true
    tuple val(meta), path("*.bootstrap") , emit: bootstrap , optional: true
```

**Review by JoseEspinosa — COMMENTED** [review]
Seems I forgot to push the button 😛

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
Very nice work! 
A couple of suggestions that should have been included in the initial PR:
- Add  stub test
- With the boostrapping test
      - Try to do the following for the outputs, I know bootstrapping is non-deterministic but the nwk file and stats file will contain the sample names in there so : [check if a string is contained](https://nf-co.re/docs/contributing/tutorials/nf-test_assertions#file-contains-check) >  [check if the file exists](https://nf-co.re/docs/contributing/tutorials/nf-test_assertions#file-exists-check) > [check if file name matches](https://nf-co.re/docs/contributing/tutorials/nf-test_assertions#snapshot-selective-file-names--content) 
      - Include stats & versions in assertAll clause

**Comment by itrujnara** [comment]
@Joon-Klaps is this what you meant? I have not been able to adapt the "file exists" example, nor do I know how to approach the one with sample name (recalculate prefix in test?). I have not yet added the stub test.

**Comment by itrujnara** [comment]
Stub test is now added

File: modules/nf-core/fastme/tests/main.nf.test [hunk 2]
```diff
@@ -110,18 +111,44 @@ nextflow_process {
...
             assertAll(
                 { assert process.success },
-                { assert process.out.nwk },
-                { assert process.out.matrix },
-                { assert process.out.bootstrap }
+                { assert process.out.nwk[0][1] =~ ".*.nwk" },
+                { assert process.out.matrix[0][1] =~ ".*.matrix.phy" },
+                { assert process.out.bootstrap[0][1] =~ ".*.bootstrap" },
+                { assert process.out.stats[0][1] =~ ".*_stat.txt" },
+                { assert process.out.versions }
```
Comment (Joon-Klaps) [inline modules/nf-core/fastme/tests/main.nf.test:127]:
```suggestion
                { assert path(process.out.nwk[0][1]).text.contains("1atx:") },
                { assert path(process.out.matrix[0][1]).text.contains("1apf") },
                { assert path(process.out.bootstrap[0][1]).text.contains("1atx:") },
                { assert path(process.out.stats[0][1]).readLines()[0..12]).match("stats_boot") },
                { assert snapshot(process.out.versions).match("versions") }
```

I meant something like this. If you look at the [output.zip](https://github.com/nf-core/modules/files/14646877/fastme_out.zip), you can see that some strings are contained across runs and we can check for it 😊. Also don't forget to include a snapshot for the versions. The current test clause, doesn't include any snapshots and it won't be written to the `main.nf.test.snap` file. 

Double check the test for errors, update snapshot and then 🚀

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]

**Comment by itrujnara** [comment]
There was a minor issue in one test (`snapshot()` missing), but I fixed it and now it's passing locally. Can you have another look @Joon-Klaps?

**Review by Joon-Klaps — APPROVED** [review]
Great work 🚀! Sorry for nitpicking.

**Pull request merged**